vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNavItem.java

```diff
@@ -221,6 +304,93 @@ public String getPath() {
         return getElement().getAttribute("path");
     }
 
+    private void doSetPath(String path) {
+        if (path == null) {
+            getElement().removeAttribute("path");
+        } else {
+            getElement().setAttribute("path",
+                    sanitizePath(updateQueryParameters(path)));
+        }
+    }
+
+    /**
+     * Gets the {@link RouteParameters} of this item.
+     *
+     * @return {@link RouteParameters} of this item
+     * @see #setRouteParameters(RouteParameters)
+     */
+    public RouteParameters getRouteParameters() {
+        return routeParameters;
+    }
+
+    /**
+     * Sets the {@link RouteParameters} of this item.
+     *
+     * @param routeParameters
+     *            the route parameters
+     */
+    public void setRouteParameters(RouteParameters routeParameters) {
```

Comment:
If we agree to only support router parameters when setting a view, this method appears to become useless, as you can always do `setPath(view, newRouterParameters)` to update the path.

Reply:
I will remove this functionality. We can always add it later if there are actual use cases and demand for it.

Reply:
Yeah, I agree. It may be a good idea to avoid introducing public APIs for which there is no high necessity.

Reply:
Removed.